tests/dedupe.js

```diff
@@ -1,7 +1,7 @@
 /* global describe, it */
 
-var assert = require('assert');
-var dedupe = require('../dedupe');
+import assert from 'assert';
```

Comment:
Would it also make more sense to use `node:` internal namespace for these built-ins? For example:

```js
import { assert } from 'node:assert'
```

Reply:
I think you mean:

```js
import { strictEqual } from 'node:assert';
```

If you do use the `node:` internal namespace, you need to account for it being added in Node.js v12.20.0, v14.13.1 in the package `engines.node` field:

https://nodejs.org/api/esm.html#esm_node_imports

I don't have a hard opinion if the `node:` internal namespace is a good idea for package authors to use yet. Up front it doesn't have a concrete benefit, but it smells like the sort of thing that might mess with people's bundler config and detection of Node.js imports for marking as external, people's regex, etc. Generally it's a good idea though, and @sindresorhus is using it in his packages and encouraging it for the wider ecosystem migrating to ESM:

https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c#how-can-i-move-my-commonjs-project-to-esm

Reply:
Since these files are reserved for the tests we are running and are not shipped in the final package I think we can actually make this change without any impact on users. I've gone ahead and used named exports for all Node internals, including the namespace.